store/src/java/com/zimbra/cs/imap/NioImapRequest.java

```diff
@@ -53,6 +53,9 @@ private void parseLiteral(byte[] b) throws IOException {
 
     private void parseCommand(String line) throws IOException, ProtocolDecoderException {
         addPart(line);
+        if (line.endsWith("\r\n")) {
+            line = line.substring(0, line.lastIndexOf('\r'));
```

Comment:
What's the problem being addressed here?  The comment says the same as the code - what is being done but not why.  https://tools.ietf.org/html/rfc3501#section-2 says this:

> All interactions transmitted by client and server are in the form of lines, that is, strings that end with a CRLF.

So this feels wrong to me.  I'm also rather puzzled why this change and the previous change affect imapd but apparently don't affect either of the embedded variants?